### PR TITLE
bump ethers-rs dependencies to onbjerg's ast

### PR DESCRIPTION
**Pull request by iFrostizz**
## Motivation

Following the discussion https://github.com/gakonst/ethers-rs/pull/1943 about checking for issues related to foundry dependency to ethers-rs ast that has been reworked by @onbjerg 
Issues seems to occur on the coverage part as expected. I am happy to refactor this once the ast is merged on the ethers-rs side.

### CODE REVIEW

**Comment by gakonst** [comment]
Nice! OK would prefer that we do the fixes on this PR against the ethers-rs branch and merge them both in lockstep, instead of merging one, then doing this PR, as it can cause breakages

**Comment by iFrostizz** [comment]
> Nice! OK would prefer that we do the fixes on this PR against the ethers-rs branch and merge them both in lockstep, instead of merging one, then doing this PR, as it can cause breakages

Can we make an ether-rs branch so I can change the PR target from my branch with the ast, update it on the foundry repo in this PR, and then do another PR as a 2nd step after merging the PR from ethers into master ?

**Comment by gakonst** [comment]
ye that works

**Comment by iFrostizz** [comment]
how does that look @gakonst ?

**Comment by iFrostizz** [comment]
Bump

**Comment by mattsse** [comment]
looks like it, so supportive of merging both

**Comment by gakonst** [comment]
@iFrostizz merged - let's make this use ethers master and good to go

**Comment by mattsse** [comment]
okay, there are way more breaking changes here than anticipated...

this PR didn't actually check against the ethers PR `rev = bb4af1c` is not the right check so this was misleading

**Comment by gakonst** [comment]
Merged the reverted PR. @iFrostizz this is why we should not try to rush things 😅

File: config/src/chain.rs [hunk 1]
```diff
@@ -97,12 +97,12 @@ impl From<Chain> for U256 {
 }
 
 impl TryFrom<Chain> for ethers_core::types::Chain {
-    type Error = ParseChainError;
+    type Error = String;
```
Comment (DaniPopes) [inline config/src/chain.rs:100]: https://github.com/foundry-rs/foundry/pull/3937/commits/322ba09a4cee500e77c49ef488ecb7037c909b53

```suggestion
    type Error = <ethers_core::types::Chain as TryFrom<u64>>::Error;
```

File: config/src/chain.rs [hunk 2]
```diff
@@ -97,12 +97,12 @@ impl From<Chain> for U256 {
...
 
     fn try_from(chain: Chain) -> Result<Self, Self::Error> {
         match chain {
             Chain::Named(chain) => Ok(chain),
-            Chain::Id(id) => id.try_into(),
+            Chain::Id(id) => ethers_core::types::Chain::try_from(id).map_err(|err| err.to_string()),
```
Comment (DaniPopes) [inline config/src/chain.rs:105]: https://github.com/foundry-rs/foundry/pull/3937/commits/322ba09a4cee500e77c49ef488ecb7037c909b53

```suggestion
            Chain::Id(id) => id.try_into(),
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]
Didn't realize you were bumping ethers here so I closed #3937

**Comment by iFrostizz** [comment]
Sorry friends but what exactly caused this issue ?
"rev = bb4af1c was not the right check so the ci results of this PR were misleading"
My main intent was to pin the dependencies of foundry to ethers as the "latest stable", then merge the ast in ethers-rs to give a little bit of time in case anything was broken on foundry side (which seems to have happened), and after fixing open a new PR in order to update foundry as well as the dependencies.

**Pull request merged**